src/serde_utils.rs

```diff
@@ -0,0 +1,107 @@
+
+//! Module for special serde serializations.
+
+pub mod btreemap {
+    //! Module for serialization of BTreeMaps because serde_json will
+    //! not serialize hashmaps with non-string keys be default.
+    #![allow(missing_docs)]
+
+    // NOTE: This module can be exactly copied to use with HashMap.
+
+    use ::std::collections::BTreeMap;
+    use serde;
+
+    pub fn serialize<S, T, U>(v: &BTreeMap<T, U>, s: S)
+        -> Result<S::Ok, S::Error> where
+        S: serde::Serializer,
+        T: serde::Serialize + ::std::hash::Hash + Eq + Ord,
+        U: serde::Serialize,
+    {
+        use serde::ser::SerializeSeq;
+
+        let mut seq = s.serialize_seq(Some(v.len()))?;
+        for pair in v.iter() {
+            seq.serialize_element(&pair)?;
```

Comment:
One slightly annoying consequence of directly serializing the tuple instead of a wrapper is that the value is serialized as array of `u8`s instead of hex like everything else.

Reply:
I think this is more than slightly annoying but it's also really unclear how to fix it.

Reply:
It might be worthwhile to duplicate this entire module, changing `U` to `Vec<u8>` and doing hex-serialization.

Reply:
Could we get away with having a `Preferred(De)Serialize` trait which is identical to the `(De)Serialize` traits and has a default impl just passing through to the serde version. But for certain types we can implement default (de)serialization methods.